include/onnxruntime/core/platform/ort_mutex.h

```diff
@@ -187,3 +188,36 @@ std::cv_status OrtCondVar::wait_for(std::unique_lock<OrtMutex>& cond_mutex,
 }
 };  // namespace onnxruntime
 #endif
+
+namespace onnxruntime {
+
+/*
+OrtSpinLock implemented mutex semantic "lock-freely",
+calling thread will not be put to sleep on blocked,
+which reduces cpu usage on context switching.
+*/
+struct OrtSpinLock {
```

Comment:
1. Do you intend to make this available to everyone including ort_mutex.h? If not, can this be contained within the threadpool header only?
2. Can the usage of this lock be made configurable? Busy waiting will consume a lot of CPU cycles and not all workloads may benefit from it.

Reply:
We don't know the types of workloads our customers are running. Hence, I would still prefer to make this configurable to avoid any surprises.

Reply:
Given the varied workload - we are migrating from heavyweight locking to lightweight, the latter incurs no os API calls and leaves the thread's state as it is. The reduction of context switch overall downsizes CPU usage on context switching - the improvement should benefit general cases, not merely a handful.